src/goto-analyzer/goto_analyzer_parse_options.cpp

```diff
@@ -131,6 +139,232 @@ void goto_analyzer_parse_optionst::get_command_line_options(optionst &options)
+    options.set_option("general-analysis", true);
+    options.set_option("non-null", true);
+    options.set_option("domain set", true);
+  }
+  else if(cmdline.isset("intervals") || cmdline.isset("non-null"))
+  {
+    // For backwards compatibility either of these on their own means show
+    options.set_option("show", true);
+    options.set_option("general-analysis", true);
+  }
+
+  if(options.get_bool_option("general-analysis") || reachability_task)
+  {
+    // Abstract interpreter choice
+    if(cmdline.isset("location-sensitive"))
+      options.set_option("location-sensitive", true);
+    else if(cmdline.isset("concurrent"))
+      options.set_option("concurrent", true);
+    else
+    {
+      // Silently default to location-sensitive as it's the "default"
+      // view of abstract interpretation.
+      options.set_option("location-sensitive", true);
+    }
+
+    // Domain choice
+    if(cmdline.isset("constants"))
+    {
+      options.set_option("constants", true);
+      options.set_option("domain set", true);
+    }
+    else if(cmdline.isset("dependence-graph"))
+    {
+      options.set_option("dependence-graph", true);
+      options.set_option("domain set", true);
+    }
+    else if(cmdline.isset("intervals"))
+    {
+      options.set_option("intervals", true);
+      options.set_option("domain set", true);
+    }
+    else if(cmdline.isset("non-null"))
+    {
+      options.set_option("non-null", true);
+      options.set_option("domain set", true);
+    }
+
+    // Reachability questions, when given with a domain swap from specific
+    // to general tasks so that they can use the domain & parameterisations.
+    if(reachability_task)
+    {
+      if(options.get_bool_option("domain set"))
+      {
+        options.set_option("specific-analysis", false);
+        options.set_option("general-analysis", true);
+      }
+    }
+    else
+    {
+      if(!options.get_bool_option("domain set"))
+      {
+        // Default to constants as it is light-weight but useful
+        status() << "Domain defaults to --constants" << eom;
+        options.set_option("constants", true);
+      }
+    }
+  }
+}
+
+/// For the task, build the appropriate kind of analyzer
+/// Ideally this should be a pure function of options.
+/// However at the moment some domains require the goto_model
+ai_baset *goto_analyzer_parse_optionst::build_analyzer(const optionst &options)
+{
+  ai_baset *domain = nullptr;
+
+  if(options.get_bool_option("location-sensitive"))
+  {
+    if(options.get_bool_option("constants"))
+    {
+      // constant_propagator_ait derives from ait<constant_propagator_domaint>
+      domain=new constant_propagator_ait(goto_model.goto_functions);
+    }
+    else if(options.get_bool_option("dependence-graph"))
+    {
+      domain=new dependence_grapht(namespacet(goto_model.symbol_table));
+    }
+    else if(options.get_bool_option("intervals"))
+    {
+      domain=new ait<interval_domaint>();
+    }
+#if 0
```

Comment:
If this isn't implemented yet, rather than `#if` ing it out, is it better to either not include this block at all, or to replace the body of the if with an UNREACHABLE or something with an appropriate message?

Reply:
It should already give an appropriate message.  I'd like to keep the #if 0 for now to show how to integrate these things.